src/components/input/Select.tsx

```diff
@@ -183,46 +190,60 @@ function SelectOption<T>({
     >
       <div
         className={classnames(
-          'w-full flex justify-between items-center gap-3',
-          'rounded py-2 px-3',
+          // Make items stretch so that all have the same height. This is
+          // important for multi-selects, where the checkbox actionable surface
+          // should span to the very edges of the option containing it.
+          'flex justify-between items-stretch gap-3',
+          'w-full relative rounded py-2 px-3',
           {
             'hover:bg-grey-1 group-focus-visible:ring': !disabled,
             'bg-grey-1 hover:bg-grey-2': selected,
           },
         )}
       >
-        {optionChildren(children, { selected, disabled })}
+        <div className="flex items-center">
+          {optionChildren(children, { selected, disabled })}
+        </div>
         {!multiple && (
-          <CheckIcon
-            className={classnames('text-grey-6 scale-125', {
-              // Make the icon visible/invisible, instead of conditionally
-              // rendering it, to ensure consistent spacing among selected and
-              // non-selected options
-              'opacity-0': !selected,
-            })}
-          />
-        )}
-        {multiple && (
-          <div
-            className={classnames('scale-125', {
-              'text-grey-6': selected,
-              'text-grey-3 hover:text-grey-6': !selected,
-            })}
-          >
-            <Checkbox
-              checked={selected}
-              checkedIcon={CheckboxCheckedFilledIcon}
-              elementRef={checkboxRef}
-              onChange={toggleValue}
-              onKeyDown={e => {
-                if (e.key === 'ArrowLeft') {
-                  e.preventDefault();
-                  optionRef.current?.focus();
-                }
-              }}
+          <div className="flex items-center">
```

Comment:
Since the wrapper is no `items-stretch` instead of `items-center`, every element that is now rendered inside of it needs to define its ow `flex` container with `items-center`, so that everything is visually centered, while the checkbox wrapper (the label) is actually 100% tall, without having to use absolute containers, which have other side effects.